src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
That would be good. We should probably just check that both indices are positive and the end index is larger then the start index. I think its reasonable to not allow negative indices although it would be valid in Python.